consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/support/TestContextBuilder.java

```diff
@@ -307,7 +309,9 @@ private static ControllerAndState createControllerAndFinalState(
                     messageValidatorFactory),
                 messageValidatorFactory),
             gossiper,
-            DUPLICATE_MESSAGE_LIMIT);
+            DUPLICATE_MESSAGE_LIMIT,
```

Comment:
would probably prefer to see the futureMessageBuffer get created here and passed in (rather than the raw values) - why? Can't answer that.

Reply:
👍 ^^^
It's cleaner, in terms of IoC and reduces number of constructors on the controller.
I'd suggest instead of passing in values, construct & pass in the objects: 
`final MessageTracker duplicateMessageTracker, final FutureMessageBuffer futureMessageBuffer`

Reply:
That is cleaner and also reduces the need the seperate constructor in the IbftController.
Done.